crates/nu-protocol/src/ast/table.rs

```diff
@@ -0,0 +1,132 @@
+use super::Expression;
+use serde::{Deserialize, Serialize};
+use std::{fmt::Debug, iter::FusedIterator};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Table {
+    exprs: Box<[Box<[Expression]>]>,
```

Comment:
What motivates the move from having the column def separate from the data expressions in the AST as well?

Seems a bit more complex to have all this, especially as you have to construct outside at once for the `Box<[T]` fun, instead of gaining access to methods on `ast::Table` to build up.

Reply:
I can split it back into two fields: one for the column headers and one for the rows, but this will increase the size of `Expr` by 8 bytes (it's not a lot but it's still +25%). Then again, it would be nice to have a different type for the column headers, since I think the column header `Expression`s can only be `Expr::String`. Splitting the columns and rows would also simplify the `Table` API, so I guess it's probably the way to go.

In that case, I wonder if it's beneficial to box `Table` in `Expr::Table`, since `Expr::Table` tends to be an uncommon expression and this would keep `Expr` at it's current size of 32 bytes.